pkg/operator/managed/handler.go

```diff
@@ -101,16 +105,90 @@ func (h *Handler) getServiceClass(service *v1alpha1.SharedService) (*v1beta1.Clu
 	}
 
 	for _, sc := range scs.Items {
-		if sc.Spec.CommonServiceClassSpec.ExternalName == svcCopy.Spec.ClusterServiceClassExternalName {
-			svcCopy.Spec.ClusterServiceClassName = sc.Name
+		if sc.Spec.CommonServiceClassSpec.ExternalName == service.GetClusterServiceClassExternalName() {
+			service.SetClusterServiceClassName(sc.Name)
 			return &sc, nil
 		}
 	}
 
-	return nil, errors.New("Could not find a matching Cluster Service Class for:" + svcCopy.Spec.ClusterServiceClassExternalName)
+	return nil, errors.New("Could not find a matching Cluster Service Class for:" + service.GetClusterServiceClassExternalName())
+}
+
+func (h *Handler) provisionService(namespace, serviceClassExternalName string, params map[string]interface{}, sc *v1beta1.ClusterServiceClass) (*v1beta1.ServiceInstance, error) {
+	parameters, err := json.Marshal(params)
+	if err != nil {
+		return nil, err
+	}
+
+	si := buildServiceInstance(namespace, serviceClassExternalName, parameters, *sc)
+	return h.serviceCatalogClient.Servicecatalog().ServiceInstances(namespace).Create(&si)
 }
 
 func (h *Handler) handleSharedServiceCreateUpdate(service *v1alpha1.SharedService) error {
+	if len(service.Spec.CurrentInstances) < service.Spec.RequiredInstances {
```

Comment:
why required instead of min

Reply:
there is both required and min, I thought the idea would be that required is based on the number of used slices, but min is defined by the service (i.e. if the service takes a really long time to provision, don't ever go below 1, even if no slices are in use).